src/phoenix/server/api/subscriptions.py

```diff
@@ -64,36 +86,79 @@ async def chat_completion(
 
         client = AsyncOpenAI()
 
-        # Loop over the input messages and map them to the OpenAI format
-
-        messages: List[ChatCompletionMessageParam] = [
-            to_openai_chat_completion_param(message) for message in input.messages
-        ]
-        chunk_contents = []
-        start_time = datetime.now()
-        async for chunk in await client.chat.completions.create(
-            messages=messages,
-            model="gpt-4",
-            stream=True,
-        ):
-            choice = chunk.choices[0]
-            if choice.finish_reason is None:
-                assert isinstance(chunk_content := chunk.choices[0].delta.content, str)
-                yield chunk_content
-                chunk_contents.append(chunk_content)
-        end_time = datetime.now()
-        async with info.context.db() as session:
-            # insert dummy data
-            trace_id = str(start_time)
-            span_id = str(end_time)
-            default_project_id = await session.scalar(
-                select(models.Project.id).where(models.Project.name == "default")
+        in_memory_span_exporter = InMemorySpanExporter()
+        tracer_provider = TracerProvider()
+        tracer_provider.add_span_processor(
+            span_processor=SimpleSpanProcessor(span_exporter=in_memory_span_exporter)
+        )
+        tracer = tracer_provider.get_tracer(__name__)
+        span_name = "ChatCompletion"
+        with tracer.start_span(
+            span_name,
+            attributes=dict(
+                chain(
+                    _llm_span_kind(),
+                    _input_value_and_mime_type(input),
+                    _llm_input_messages(input.messages),
+                )
+            ),
+        ) as span:
+            chunks = []
+            chunk_contents = []
+            role: Optional[str] = None
+            async for chunk in await client.chat.completions.create(
+                messages=(to_openai_chat_completion_param(message) for message in input.messages),
+                model="gpt-4",
+                stream=True,
+            ):
+                chunks.append(chunk)
+                choice = chunk.choices[0]
+                delta = choice.delta
+                if role is None:
+                    role = delta.role
+                if choice.finish_reason is None:
+                    assert isinstance(chunk_content := delta.content, str)
+                    yield chunk_content
+                    chunk_contents.append(chunk_content)
+            span.set_status(StatusCode.OK)
```

Comment:
do we need error handling?

Reply:
I think it's handled by the context manager.

Reply:
Will make sure when I add tests.